include/boost/http_proto/request.hpp

```diff
@@ -36,6 +35,120 @@ class BOOST_SYMBOL_VISIBLE
     request(
         core::string_view s);
 
+    /** Constructor
+
+        Construct a request container which allocates
+        `storage_size` bytes for storing the header.
+        Attempting to grow the container beyond
+        `storage_size` bytes will result in an exception.
+        The storage is also used internally to store
+        instances of an implementation-defined type.
+        `storage_size` will be aligned accordingly
+        (currently the alignment requirement is 4).
+
+        <br/>
+
+        This constructor is useful when an upper-bound size
+        of the request is known ahead of time and we want
+        to prevent reallocations.
+
+        <br/>
+
+        Passing an initial storage size of `0` does not
+        throw and the maximum capacity is set to an
+        implementation-defined limit observable via
+        `max_capacity_in_bytes()`.
+
+        @param storage_size The initial and final size of
+        the storage allocated to hold both the serialized
+        header and instances of a per-field
+        implementation-defined type.
```

Comment:
You don't have to be verbose and repeat information which was already given in the description. Better to only explain things once, this way if you ever have to change the javadoc you can't forget to edit it everywhere.